File: server/models/userModel.js
import mongoose from "mongoose";

const userSchema = new mongoose.Schema(
  {
    uid: {
      type: String,
      required: true,
      unique: true, // Firebase UID
    },
    phone: {
      type: String,
      default: null,
    },
    // address: {
    //   home: {
    //     type: String,
    //     default: null,
    //   },
    //   work: {
    //     type: String,
    //     default: null,
    //   },
    // },
    address: {
      type: [mongoose.Schema.Types.Mixed],
      default: [],
    },
    vton_image: {
      type: String,
      default: null, // URL or file path
    },
    type: {
      type: String,
      enum: ["normal", "seller"],
      default: "normal",
    },
  },
  { timestamps: true }
);
const User = mongoose.model("User", userSchema);
export default User;

File: android/app/src/main/java/com/rohan/fablefit/ui/model/UserModel.kt
package com.rohan.fablefit.ui.model

import com.google.gson.annotations.SerializedName

data class UserModel(
    @SerializedName("_id")
    val id: String = "",

    val uid: String = "",

    @SerializedName("vton_image")
    val vtonImage: String? = null,

    val type: String = "normal",

    val phone: String? = null,

    val address: List<String> = emptyList()
)